### ISSUE TITLE
gRPC: Determine what to do with response data to effectively define a response wrapper

### ISSUE THREAD

**Issue by fallwith** [newrelic/newrelic-ruby-agent#1240]
Given the binary data format, we can’t exactly match what we do with HTTP clients. Let’s review what other agents are doing.

See `lib/new_relic/agent/instrumentation/grpc/client/response_wrapper.rb` for the placeholder class.

**Comment by kaylareopelle** [comment]
TL;DR - I don't think we need a response wrapper for gRPC Clients and can remove all response wrapper boilerplate

## Findings 
* The response status will either be 0/OK, or it will raise an error before we can get to the point of recording the response. If the status is OK, it isn't returned as part of the response object. Instead, nil will be the value for status-related data.
* The noticed error includes all the relevant information (code, status, message, stacktrace) 
* The object that is returned on a successful response is the message object created by the proto (in the Route Guide example, something like `Routeguide::Point`). I don't imagine customers expect these objects to be sent up to New Relic, and it may be a security issue if they are sent up. I'm happy to adjust the ResponseWrapper for this if decide otherwise as a team.
* There isn't a response attribute in the gRPC agent-spec internal implementation guide.
* ResponseWrappers traditionally returned status codes. If the status code is anything but OK, GRPC stops execution and shuts down the client. I don't know if we necessarily want/need to have OK preserved as an attribute in that case. 
* The other procedure included in `NewRelic::Agent::Segment#process_response_headers` is `#read_response_headers`, which seems to be CAT-specific. 
    * https://github.com/newrelic/newrelic-ruby-agent/blob/dev/lib/new_relic/agent/transaction/external_request_segment.rb#L67-L90 
    * **Question: Do we want to use this for anything else? Do we even need this for anything else? Should this be marked for deprecation whenever we decide to shut down CAT?** 

If the work hasn't begun on gRPC Server Response Wrappers, I'd be happy to take that on. I think I have the groundwork understanding for it.

## Next Steps 
If this approach sounds good, I'll get together a PR that removes ResponseWrappers from gRPC Client!
